### ISSUE TITLE
Connect-Graph fails with "file not found" error when using CBA in 0.9.0 and 0.9.1

### ISSUE THREAD

**Issue by KirkMunro** [microsoftgraph/msgraph-sdk-powershell#365]
We have had scripts running in our platform using CBA for authentication against Graph with version 0.7.1 for a little while with no issues. Since we upgraded to 0.9.1 (and I'm pretty sure 0.9.0 before that as well), we have started seeing errors from Connect-Graph, as shown below. This is a regression. Did something change with CBA login logic or requirements in the 0.9.0 or 0.9.1 updates?

FYI, if I run Connect-Graph locally in an interactive session using the same CBA credentials, it works. Our platform runs scripts unattended though, and those seem to be failing, so something is definitely wrong that needs investigation.

```none
Error details:

PSMessageDetails :
Exception : System.Security.Cryptography.CryptographicException: The system cannot find the file specified.
at Microsoft.Graph.PowerShell.Authentication.Cmdlets.ConnectGraph.ProcessRecord()
at System.Management.Automation.CommandProcessor.ProcessRecord()
TargetObject :
CategoryInfo : NotSpecified: (:) [Connect-Graph], CryptographicException
FullyQualifiedErrorId : System.Security.Cryptography.CryptographicException,Microsoft.Graph.PowerShell.Authentication.Cmdlets.ConnectGraph
ErrorDetails :
InvocationInfo : System.Management.Automation.InvocationInfo
ScriptStackTrace : at Invoke-CbScript, <No file>: line 61
at <ScriptBlock>, <No file>: line 326
PipelineIterationInfo : {}

Exception details:

Message : The system cannot find the file specified.
Data : {}
InnerException :
TargetSite : Void ProcessRecord()
StackTrace : at Microsoft.Graph.PowerShell.Authentication.Cmdlets.ConnectGraph.ProcessRecord()
at System.Management.Automation.CommandProcessor.ProcessRecord()
HelpLink :
Source : Microsoft.Graph.Authentication
HResult : -2147024894

**Comment by peombwa** [comment]
@KirkMunro Thanks for reporting this. No change has been made to CBA logic since the release of 0.7.1.

Can you share the following details with us for us to repro this:
- $PSVersionTable
- Is this running on Windows server?
- Are you using a self-signed certificate? If so, was it created via `New-SelfSignedCertificate`.
- Are you using an ADFS 2019 or CNG certificate?

**Comment by KirkMunro** [comment]
* Windows PowerShell 5.1. I'll get the specific details tomorrow.
* Two different systems: Windows Server 2019 and Windows 10. I'll get the winver output tomorrow.
* Yes, we're using a self-signed certificate created via New-SelfSignedCertificate.
* I'm not sure? We created our certificate using a script that looked something like this:

```powershell
$tenantDnsName = 'example.onmicrosoft.com'
$cert = New-SelfSignedCertificate -DnsName $tenantDnsName  -CertStoreLocation Cert:\CurrentUser\My -FriendlyName 'Cloudbridge Proxy Certificate' -Provider 'Microsoft Enhanced RSA and AES Cryptographic Provider' -HashAlgorithm SHA256
$documentsFolder = [Environment]::GetFolderPath('MyDocuments')
$certificatePath = Join-Path -Path $documentsFolder -ChildPath 'cloudbridge.cer'
$null = Export-Certificate -Cert $cert -FilePath $certificatePath
```

**Comment by KirkMunro** [comment]
After more testing this morning, I was able to reproduce the issue with 0.7.1 as well, so it's not a regression. We just hadn't come across the issue until we upgraded to 0.9.1.

We connect to Graph successfully via CBA with both 0.7.1 and 0.9.1 with a script run through our Windows service, but only if the user is logged into the system. We get the error reported above if we try to connect via an unattended script run through our Windows service when the user is not logged into the system. It probably has something to do with how we run the service, impersonation, and the user profile not being accessible to whatever component is handling the CBA connection (which would explain a "not found" error -- the authentication logic is unable to find the certificate from the thumbprint provided). We get the same error from the MicrosoftTeams module when we try to connect using CBA under the same circumstances. With that being the case, I'm closing this issue while I try to pinpoint what exactly is causing the problem here.
